bi-att-flow-dev/basic/model_nqa.py

```diff
@@ -34,33 +36,51 @@ def __init__(self, config, scope, rep=True):
         N, M, JX, JQ, VW, VC, W = \
             config.batch_size, config.max_num_sents, config.max_sent_size, \
             config.max_ques_size, config.word_vocab_size, config.char_vocab_size, config.max_word_size
-        self.x = tf.placeholder('int32', [N, None, None], name='x')
+
+        # Batch major format
+
+        # Context
+        self.x = tf.placeholder('int32', [N, None, None], name='x') # going to be [N, M, JX] i.e. [batch size, max sentences, max words]
         self.cx = tf.placeholder('int32', [N, None, None, W], name='cx')
         self.x_mask = tf.placeholder('bool', [N, None, None], name='x_mask')
-        self.q = tf.placeholder('int32', [N, None], name='q')
+
+        # Query
+        self.q = tf.placeholder('int32', [N, None], name='q') # going to be [N, JQ] i.e. [batch size, max words]
         self.cq = tf.placeholder('int32', [N, None, W], name='cq')
         self.q_mask = tf.placeholder('bool', [N, None], name='q_mask')
+
+        # Outputs
         self.y = tf.placeholder('bool', [N, None, None], name='y')
         self.y2 = tf.placeholder('bool', [N, None, None], name='y2')
+
+        # Misc
         self.wy = tf.placeholder('bool', [N, None, None], name='wy')
         self.is_train = tf.placeholder('bool', [], name='is_train')
         self.new_emb_mat = tf.placeholder('float', [None, config.word_emb_size], name='new_emb_mat')
         self.na = tf.placeholder('bool', [N], name='na')
 
+        # New placeholders for our case
+        self.decoder_inputs = tf.placeholder('int32', [N, None], name='decoder_inputs') # [batch_size, max words]
+        self.target_sequence_length = tf.placeholder(shape=(N,), dtype=tf.int32, name='target_sequence_length') # batch_size
```

Comment:
could not find the  code where thee are being assigned?

Reply:
Haan yahi to karna hai hum dono ko mil ke 😃